backend/hexa/pipeline_templates/schema/mutations.py

```diff
@@ -9,9 +11,35 @@
 from hexa.user_management.models import User
 from hexa.workspaces.models import Workspace
 
+logger = logging.getLogger(__name__)
+
 pipeline_template_mutations = MutationType()
 
 
+def auto_update_pipelines_from_template_version(
+    template_version: PipelineTemplateVersion,
+):
+    template = template_version.template
+
+    pipelines_to_update = Pipeline.objects.filter(
+        source_template=template,
+        auto_update_from_template=True,
+        deleted_at__isnull=True,
+    ).select_related("workspace")
+
+    for pipeline in pipelines_to_update:
+        try:
+            template_version.create_pipeline_version(
+                workspace=pipeline.workspace, pipeline=pipeline
+            )
+        except Exception as e:
+            logger.error(
+                f"Failed to auto-update pipeline '{pipeline.name}' in workspace '{pipeline.workspace.name}': {e}",
+                exc_info=True,
+            )
+            raise
```

Comment:
Hm wait, didn't we decide we'd do everything in a DB transaction and fail the template version update if some pipeline fails its auto-update? @nazarfil